internal/storage/bucket_handle_test.go

```diff
@@ -158,9 +158,26 @@ func (t *BucketHandleTest) TestDeleteObjectMethodWithMissingObject() {
 	error := t.bucketHandle.DeleteObject(context.Background(),
 		&gcs.DeleteObjectRequest{
 			Name:                       missingObjectName,
-			Generation:                 0,
+			Generation:                 TestObjectGeneration,
 			MetaGenerationPrecondition: nil,
 		})
 
 	AssertEq("storage: object doesn't exist", error.Error())
 }
+
+func (t *BucketHandleTest) TestStatObjectMethodWithValidObject() {
+	_, error := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: TestObjectName,
+		})
+
+	AssertEq(nil, error)
+}
+
+func (t *BucketHandleTest) TestStatObjectMethodWithMissingObject() {
+	_, error := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: missingObjectName,
+		})
+	AssertEq("gcs.NotFoundError: storage: object doesn't exist", error.Error())
```

Comment:
can we do error type comparision instead of string comparision.

Reply:
fixed